File: 3262-find-polygon-with-the-largest-perimeter/find-polygon-with-the-largest-perimeter.java
class Solution {
    public long largestPerimeter(int[] nums) {
        Arrays.sort(nums);
        int n = nums.length;
        long[] arr = new long[n];
        long sum = 0L;
        for(int i=0; i<n; i++) {
            sum += nums[i];
            arr[i] = sum;
        }
        long res= -1L;
        for(int i=2; i<n; i++) {
            if (arr[i-1] > nums[i]) {
                res = arr[i];
            }
        }
        return res;
    }
}

File: 3262-find-polygon-with-the-largest-perimeter/find-polygon-with-the-largest-perimeter.py
class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        # sort the nums
        nums.sort()
        # find the prefix sum
        prefix_sum = [0]*len(nums)
        sum_so_far = 0
        for i in range(len(nums)):
            sum_so_far += nums[i]
            prefix_sum[i] = sum_so_far
        max_val = -1
        for i in range(2, len(nums)):
            current_elem = nums[i]
            left_sum = prefix_sum[i] - current_elem
            if left_sum > current_elem:
                max_val =  prefix_sum[i]
        return max_val